.github/validate_robopages.yml

```diff
@@ -0,0 +1,115 @@
+name: Validate Contributions
+
+on:
+  pull_request:
+    paths:
+      - '**.yml'
+      - '!.github/**'
+    branches:
+      - main
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
+      #issues: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@c47758b77c9736f4b2ef4073d4d51994fabfe349 # 3.7.1
+
+      - name: Validate Contribution Files
+        id: robopages-validation
+        continue-on-error: true
+        run: |
+          validate_file() {
+            local file="$1"
+
+            if [[ ! "$file" =~ ^[a-zA-Z0-9_\-./]+\.yml$ ]]; then
+              echo "Invalid file path characters: $file"
+              return 1
+            }
+
+            if [[ "$file" == *"../"* ]]; then
+              echo "Directory traversal attempt detected: $file"
+              return 1
+            }
+
+            docker pull dreadnode/robopages:latest@sha256:2e355b1c46682af2fb586648016c7a28f113a407763b37c74251ae2050b396bb # f65195e7aaf5
```

Comment:
why not just latest?

Reply:
i was pinning sha's and commit hashes across the board but a fair point and removed